src/main/kotlin/com/celuveat/review/adapter/out/persistence/entity/CustomReviewRepositoryImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.celuveat.review.adapter.out.persistence.entity
+
+import com.celuveat.restaurant.adapter.out.persistence.entity.RestaurantJpaEntity
+import com.celuveat.review.adapter.`in`.rest.request.ReadReviewSortCondition
+import com.linecorp.kotlinjdsl.support.spring.data.jpa.repository.KotlinJdslJpqlExecutor
+import org.springframework.data.domain.Pageable
+import org.springframework.data.domain.Slice
+import org.springframework.data.domain.SliceImpl
+import org.springframework.stereotype.Repository
+
+@Repository
+class CustomReviewRepositoryImpl(
+    private val executor: KotlinJdslJpqlExecutor,
+) : CustomReviewRepository {
+    override fun findAllByRestaurantId(
+        restaurantsId: Long,
+        onlyPhotoReview: Boolean,
+        sort: ReadReviewSortCondition,
+        page: Pageable,
+    ): Slice<ReviewJpaEntity> {
+        val findSlice = executor.findSlice(page) {
+            select(
+                entity(ReviewJpaEntity::class),
+            ).from(
+                entity(ReviewJpaEntity::class),
+            ).whereAnd(
+                path(ReviewJpaEntity::restaurant)
+                    .path(RestaurantJpaEntity::id)
+                    .eq(restaurantsId),
+                if (onlyPhotoReview) {
+                    path(ReviewJpaEntity::hasPhoto).eq(true)
+                } else {
+                    null
+                },
```

Comment:
true/false에 따라 null을 사용 해야 하면, `takeIf`로 표현할 수 있을 것 같아요!
```suggestion
takeIf { onlyPhotoReview }?.let { path(ReviewJpaEntity::hasPhoto).eq(true) },
```